include/ofi_mr.h

```diff
@@ -327,6 +327,30 @@ bool ofi_mr_cache_flush(struct ofi_mr_cache *cache, bool flush_lru);
 
 int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct fi_mr_attr *attr,
 			struct ofi_mr_entry **entry);
+
+enum ofi_mr_cache_status{
+	OFI_MR_CACHE_STATUS_NORMAL                = 0,
+	OFI_MR_CACHE_STATUS_EXCEEDS_COUNT         = (1 << 0),
+	OFI_MR_CACHE_STATUS_EXCEEDS_SIZE          = (1 << 1),
+	OFI_MR_CACHE_STATUS_FLUSH_ENTRIES_WAITING = (1 << 2),
+};
+
+static inline unsigned int ofi_mr_cache_status_evaluate(struct ofi_mr_cache * cache)
+{
+	/* mm_lock mutex must be held during this call */
+
+	unsigned int	result = OFI_MR_CACHE_STATUS_NORMAL;
+
+	if (cache->cached_cnt  > cache_params.max_cnt)
+		result |= OFI_MR_CACHE_STATUS_EXCEEDS_COUNT;
+	if (cache->cached_size > cache_params.max_size)	
+		result |= OFI_MR_CACHE_STATUS_EXCEEDS_SIZE;
+	if (!dlist_empty(&cache->flush_list))		
+		result |= OFI_MR_CACHE_STATUS_FLUSH_ENTRIES_WAITING;
+
+	return result;
```

Comment:
If we're only concerned with result being non-zero, why add the overhead of additional checks?

Based on the changes, I would convert the function to a bool and rename to ofi_mr_cache_need_flush().  if there's really a need to return a set of flags, I would rename to ofi_mr_cache_flush_flags() and justify why we want that (unused) overhead?

Reply:
This goes back to general debugging. If we return false or true, we lose any information regarding why the decision was made. 

I understand the need for a fast path, here but memory registration is only on the critical path for the first registration of a memory region for local registration, and on reading the CQ entries for completion generation after the TXs are complete. 

My first glance at this says the additional information provided out weighs any minor impacts by having 1 or two branches in a short function anyway.